src/index.js

```diff
@@ -6,49 +6,52 @@ import jwt from "express-jwt";
 import http from "http";
 import morgan from "morgan";
 import api from "./api";
-import config from "./config";
-import reminderScheduler from './services/scheduler'
-
-let app = express();
-app.server = http.createServer(app);
-
-// logger
-app.use(morgan("dev"));
-
-// 3rd party middleware
-app.use(
-  cors({
-    exposedHeaders: config.corsHeaders,
-  })
-);
-
-app.use(bodyParser.urlencoded({ extended: true }));
-
-app.use(
-  bodyParser.json({
-    limit: config.bodyLimit,
-  })
-);
-
-reminderScheduler.start();
-
-app.use(
-  jwt({
-    secret: config.jwtConfig.jwtSecret,
-    algorithms: ["HS256"],
-  }).unless({
-    path: [
-      "/api/twilio/reply", // Authentication handled by Twilio middleware
-      "/api/ping", // Open health check
-      "/api/auth", // Used to login
-    ],
-  })
-);
-
-app.use("/api", api({ config }));
-
-app.server.listen(config.port, () => {
-  console.log(`Started on port ${app.server.address().port}`);
-});
-
-export default app;
+import { config, validateConfig } from "./config";
+import reminderScheduler from "./services/scheduler";
+
+class Application {
+  constructor() {
+    this.app = express();
+  }
+
+  start() {
+    validateConfig();
```

Comment:
This will display the config error and won't start but do we want it to be because of the exception or because of the error? For example, if you add a try/catch block around validate config you would see
```
Error: "twilioConfig.number" is required
    at module.exports.validateConfig (/media/wwan/git/sanctuary-patient-api/src/config.js:47:20)
    at Application.start (/media/wwan/git/sanctuary-patient-api/src/index.js:20:7)
    at Object.<anonymous> (/media/wwan/git/sanctuary-patient-api/src/index.js:63:13)
    at Module._compile (node:internal/modules/cjs/loader:1095:14)
    at Module._compile (/media/wwan/git/sanctuary-patient-api/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1147:10)
    at Object.newLoader [as .js] (/media/wwan/git/sanctuary-patient-api/node_modules/pirates/lib/index.js:104:7)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[nodemon] app crashed - waiting for file changes before starting...
```
where right now we see:
```
/media/wwan/git/sanctuary-patient-api/src/config.js:50
  if (error) throw new Error(error.message);
             ^

Error: "twilioConfig.number" is required
    at module.exports.validateConfig (/media/wwan/git/sanctuary-patient-api/src/config.js:47:20)
    at Application.start (/media/wwan/git/sanctuary-patient-api/src/index.js:18:7)
    at Object.<anonymous> (/media/wwan/git/sanctuary-patient-api/src/index.js:57:13)
    at Module._compile (node:internal/modules/cjs/loader:1095:14)
    at Module._compile (/media/wwan/git/sanctuary-patient-api/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1147:10)
    at Object.newLoader [as .js] (/media/wwan/git/sanctuary-patient-api/node_modules/pirates/lib/index.js:104:7)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)

Node.js v17.0.1
```

Reply:
If we add `try/catch` block, the problem is, it will continue the program.
But if we do not run the app at all, I think it will be more helpful for developers to figure out why.
What do you think?